src/math/mat4.js

```diff
@@ -271,6 +271,75 @@ Object.assign(Mat4.prototype, {
         return this;
     },
 
+    /**
+     * @function
+     * @name pc.Mat4#mulTransform2
```

Comment:
Just a suggestion, but would `mulAffine2` be better naming?

Reply:
Also, I'm wondering if this needs to be public API initially since it's only used internally. Thoughts?

Reply:
I started with mulAffine2 yesterday but wasn't sure it'd be accessible to people. I'm happy to go back to that.

I'd like this to be a public API - this should be the commonly used matrix multiply function to use in most cases. But I agree most common use would be in the engine - end users should usually just use Vec3 + quaternion.

Reply:
renamed.